kyurikotpq/cycleos-rn
# feat: Compassionate Self-Care Suggestions

Opening PR prematurely to motivate myself again

# Description
This PR introduces one compassionate suggestion to nudge the user to rest when healthy baselines are breached.

For the suggestion to show up, the user must:
- Have a Total Sleep Duration of less than 7 hours (< 420 minutes) the previous night, across all sleep sessions (e.g. if the user awoke at 2am, stayed awake till 3am, but managed to fall back asleep, the wearable will likely record two sleep sessions; CycleOS will count the total sleep time across these two sessions)
- Rate a negative symptom from the Energy category (i.e. Tired, Exhausted)
- Not have dismissed the suggestion before for the current day

The suggestion card encourages the user to schedule a nap. Pressing No dismisses the card for the day (tracked using SecureStore, which stores the dismissal time). Pressing Yes will direct the user to the system UI for adding a new Calendar event. 

Other mini details added:
- Card in the Agenda screen encouraging logging of symptoms ("Daily Check In Card" (DCIC))
- Dialog thanking the user for logging their symptoms and asking if they want to check their insights
- Motivating quotes (morning) and compassionate quotes (evening) for the title of DCIC
- A custom theme with one accent color

Related Issues: 
- Closes #10 in 4c92b96f3889f40b909ab7b80aa03c0ec65072fb
- Checks off 2 items in #9 (Color bar and legend added)

# Type of Change
Please delete options that are not relevant and provide a brief explanation:

- [ ] Documentation update (adds or updates documentation only)

# Checklist
Ensure your pull request fulfills the following requirements:

- [x] I have followed the branch naming convention (<your-username>/<feature-or-task>).
- [ ] My code adheres to the project's code style guidelines.
- [x] I have commented my code where necessary, particularly in hard-to-understand areas.
- [ ] I have added relevant tests for my changes (if applicable).

# How Has This Been Tested?
Tested manually on Android 15, Google Pixel 8a

Flow for having the naptime show up:
- Sleep less than 7h
- Make sure poor sleep was recorded in HealthConnect (in my case, via Samsung Health)
- Log a negative Energy symptom (Tired or Exhausted) 
- NapOrMeditateCard shows up
- Press Yes -> Google Calendar Event shows up
- Press No -> Card gets dismissed (disappears)

# Screenshots (if applicable)
![photo_6244393737396537075_y](https://github.com/user-attachments/assets/e28fe70a-2a73-4d99-b4ea-0550a1fbf07f)

# Additional Notes
Future suggestions to include:
- Checking for sleep fragmentation, later than usual sleep start time, and wakefulness (instead of just sleep duration)
- Checking step count wrt to daily step targets for the past 3 days. If the number of days where `steps < target count` is 2 or more, encourage the user to make time for walking
- Checking total exercise time for the past week. If it's too low (<120 minutes?), encourage the user to make time for exercise.

Future extensions:
- Check one's calendars (determined by the user) for "free space". Provide a list of suggestions for nap times (free space of >= 25 mins before 3pm) and meditation times (free space of >= 10mins any time)
